### DISCUSSION TITLE
Can't open file on Samba drive (Ubuntu 22.10)

### DISCUSSION THREAD

**Discussion by LarryM04** [smplayer-dev/smplayer#588] [General]
I have a pair of boxes that had been on Ubuntu 22.04.  One box is the server for a Samba drive, the other is the client.  SMPlayer opened and played video files on this drive just fine.  Then I upgraded the client box to Ubuntu 22.10, now SMPlayer can't read/open any video files on the shared drive.  Files on the local drive play fine.  The built-in video play "Video" plays videos on this drive.

I've tried accessing/playing the video files two different ways, one by using the file manager to navigate to the file and double clicking on it.  SMPlayer starts, pops up an error message and closes -2.  The log says it cannot open the file.  If I do "Open|File" on the SMPlayer bar, then click "Computer" then "/" a small set of directories open: bin, etc, home, lib, lib32, meta, snap, usr, and var - the disk is mount under /smb.

Why can't SMPlayer see the video files on this Samba drive?  SMPlayer is version 22.7.0 (revision 10091)

**Comment by smplayer-dev** [comment]
Are you using SMPlayer from a snap package?
Applications from snap packages run with restrictions, you may need to give SMPlayer some permissions.

**Reply by LarryM04** [reply]
I did whatever the apt-get command line that was given on the "Download" page.  What/how do I give the permissions?  The files, and directory tree to them are all read-write for all.  Just tried running "sudo smplayer" and indeed it was able to access and play the files on the Samba drive.  So what executable needs adjustment, and to what?

**Comment by smplayer-dev** [comment]
if you installed it with apt, then I think that's not the problem. What distro are you using?

**Reply by LarryM04** [reply]
Its Ubuntu 22.10, installed on a fresh drive (not an upgrade).

**Comment by LarryM04** [comment]
Is there any hope for this?

**Comment by smplayer-dev** [comment]
I'm afraid I didn't have the time yet to look into this issue.
Anyway, is the option "use the system native file dialog" (preferences -> interface) enabled?

**Comment by LarryM04** [comment]
Yes, its enabled.  You'd previously mentioned permissions not being set 
if I'd use snap to install.  Can you tell me what to look at?  Maybe the 
apt install missed it?

thanks - Larry

**Comment by smplayer-dev** [comment]
https://ubuntu.com/blog/a-guide-to-snap-permissions-and-interfaces

**Comment by LarryM04** [comment]
I don't notice any problems:
***@***.***: snap interfaces smplayer
Slot                     Plug
smplayer:mpris           -
:audio-playback          smplayer
:desktop                 smplayer
:desktop-legacy          smplayer
:home                    smplayer
:network                 smplayer
:network-bind            smplayer
:opengl                  smplayer
:optical-drive           smplayer
:screen-inhibit-control  smplayer
:unity7                  smplayer
:wayland                 smplayer
:x11                     smplayer
-                        smplayer:alsa
-                        smplayer:dvb
-                        smplayer:jack1
-                        smplayer:mount-observe
-                        smplayer:network-control
-                        smplayer:network-manager
-                        smplayer:pulseaudio
-                        smplayer:removable-media

'snap interfaces' is deprecated; use 'snap connections'.
***@***.***: snap connections smplayer
Interface               Plug Slot                     Notes
alsa                    smplayer:alsa -                        -
audio-playback          smplayer:audio-playback :audio-playback          -
desktop                 smplayer:desktop :desktop                 -
desktop-legacy          smplayer:desktop-legacy :desktop-legacy          -
dvb                     smplayer:dvb -                        -
home                    smplayer:home :home                    -
jack1                   smplayer:jack1 -                        -
mount-observe           smplayer:mount-observe -                        -
mpris                   - smplayer:mpris           -
network                 smplayer:network :network                 -
network-bind            smplayer:network-bind :network-bind            -
network-control         smplayer:network-control -                        -
network-manager         smplayer:network-manager -                        -
opengl                  smplayer:opengl :opengl                  -
optical-drive           smplayer:optical-drive :optical-drive           -
pulseaudio              smplayer:pulseaudio -                        -
removable-media         smplayer:removable-media -                        -
screen-inhibit-control  smplayer:screen-inhibit-control 
:screen-inhibit-control  -
unity7                  smplayer:unity7 :unity7                  -
wayland                 smplayer:wayland :wayland                 -
x11                     smplayer:x11 :x11                     -
***@***.***:

**Comment by LarryM04** [comment]
Wondering if it was a file permission thing, I did chmod 4755 and owner to root on /usr/bin/smplayer ... still couldnt access files

**Reply by wvdberg** [reply]
would it not be the permissions of your mounted files that are the problem?

```
ls -lah /smb
```

you could conversely try running smplayer as root? (edit: as a diagnostic step)

```
sudo smplayer 
```

**Comment by LarryM04** [comment]
Permissions are wide open:
```
larry@asrock: pwd
/smb
larry@asrock: ls -l
total 12
drwxrwxr-x 4 larry larry 4096 Oct 23 14:24 asus
drwxrwxr-x 5 larry larry 4096 Oct 23 14:25 fs1
drwxrwxr-x 6 larry larry 4096 Oct 23 14:26 Helios4
larry@asrock: ls -l Helios4
total 0
drwxr-xr-x 2 larry larry 0 May  5  2022 Misc
drwxr-xr-x 2 larry larry 0 Sep 21 14:43 Movies
drwxr-xr-x 2 larry larry 0 Sep 29  2020 Other
drwxr-xr-x 2 larry larry 0 Nov 12 15:16 TV
larry@asrock: ls -l Helios4/Movies
total 0
drwxr-xr-x 2 larry larry 0 Feb  2  2022 'Dog Training'
drwxr-xr-x 2 larry larry 0 Sep 21 14:43  DVDs
drwxr-xr-x 2 larry larry 0 Feb 23  2018 'Eddie Izzard'
drwxr-xr-x 2 larry larry 0 Jan 19  2020  Fishing
drwxr-xr-x 2 larry larry 0 Feb 23  2018  Gallagher
drwxr-xr-x 2 larry larry 0 Mar 11  2022  TaiChi
drwxr-xr-x 2 larry larry 0 Dec 12 14:16  Torrents
larry@asrock: ls -l Helios4/Movies/Torrents
total 340484096
-rwxr-xr-x 1 larry larry 1316138365 Dec 13  2019  6.Underground.2019.WEBRip.x264-ION10.mp4
-rwxr-xr-x 1 larry larry     150079 Dec 13  2019  6.Underground.2019.WEBRip.x264-ION10.srt
drwxr-xr-x 2 larry larry          0 Mar 23  2019 'Abbott and Costello'
-rwxr-xr-x 1 larry larry  531093932 Jul 25  2018  Abbott.And.Costello.In.The.Foreign.Legion.1950.DVDRip.480p.x264.shahid4u.com.mp4
-rwxr-xr-x 1 larry larry  733026304 Aug 31  2018 'Abbott and Costello Meet The Invisible Man (1951).avi'

```

BUT!  Your other idea, sudo smplayer - uncovers another bit of a clue.  Turns out starting smplayer from the terminal command line (with out without the sudo) and then doing File|Open successfully plays the file - as does starting smplayer from selecting the 9-dots, then smplayer.  Its only when double clicking a file from the file manager that fails.

So what is double clicking doing different than just opening smplayer?

**Comment by LarryM04** [comment]
I've noticed something - When I start SMPlayer from the command line, then do "Open|File" a normal looking dialog box opens and I can select any file (even on the Samba drives) and the file plays normally.   But - if I start SMPlayer by double clicking a file on the Samba drives, then do the "Open|File" a completely different looking dialog box displays which shows just "Computer" and "64" on the left hand panel.  No idea what "64" is.  Selecting either does nothing.  ???

**Reply by LarryM04** [reply]
Further - if I single click a video file in Nautilus, then select "Open with" and select SMPlayer the video plays!  Something is wrong with Nautilus not SMPlayer
